src/tests/suite/daos_obj.c

```diff
@@ -750,6 +809,107 @@ io_overwrite(void **state)
 	io_overwrite_large(state, oid);
 }
 
+static void
+io_rewritten_same_array_with_large_small_size_verify_poolsize(void **state)
+{
+	test_arg_t		*arg = *state;
+	struct ioreq		req;
+	daos_obj_id_t		oid;
+	daos_pool_info_t	pinfo;
+	char			*ow_buf;
+	char			*fbuf;
+	const char		dkey[] = "dkey";
+	const char		akey[] = "akey";
+	daos_size_t		size = 4 * 1024; /* record size */
+	int			buf_idx; /* overwrite buffer index */
+	int			rx_nr; /* number of record extents */
+	int			record_set;
+	int			rc;
+	daos_size_t		nvme_initial_size;
+	daos_size_t		nvme_current_size;
+
+	/* choose random object */
+	oid = dts_oid_gen(dts_obj_class, 0, arg->myrank);
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	/* Alloc and set buffer to be a sting*/
+	D_ALLOC(ow_buf, size);
+	assert_non_null(ow_buf);
+	dts_buf_render(ow_buf, size);
+	/* Alloc the fetch buffer */
+	D_ALLOC(fbuf, size);
+	assert_non_null(fbuf);
+	memset(fbuf, 0, size);
+
+	/* Get the pool info at the beginning */
+	rc = pool_storage_info(state, &pinfo);
+	assert_int_equal(rc, 0);
+	nvme_initial_size = pinfo.pi_space.ps_space.s_free[1];
+
+	/* Set and verify the full initial string in first transaction */
+	rx_nr = size / OW_IOD_SIZE;
+	/* Insert the initial 4K record which will go through NVMe */
+	insert_single_with_rxnr(dkey, akey, /*idx*/0, ow_buf, OW_IOD_SIZE,
+				rx_nr, DAOS_TX_NONE, &req);
+	/* Lookup the initial 4K record */
+	lookup_single_with_rxnr(dkey, akey, /*idx*/0, fbuf, OW_IOD_SIZE, size,
+				DAOS_TX_NONE, &req);
+	/* Verify the 4K data */
+	assert_memory_equal(ow_buf, fbuf, size);
+
+	/**
+	*Get the pool storage information
+	*/
+	rc = pool_storage_info(state, &pinfo);
+	assert_int_equal(rc, 0);
+	nvme_current_size = pinfo.pi_space.ps_space.s_free[1];
+
+	/**
+	* Verify data written on NVMe by comparing the NVMe Free size.
+	*/
+	if (nvme_current_size > nvme_initial_size - size) {
```

Comment:
Disable aggregation before this test? And I'm wondering why "nvme_current_size" isn't equal to "nvme_initial_size - size" here?

Reply:
Disable the Aggregation for this test.
Default object class is OC_RP_2G1 so for this data set NVMe free size is going to reduce (8K) I am checking to make sure it's at least decreasing the requested base record size 4K.